climada/engine/test/test_impact_calc.py

```diff
@@ -135,6 +135,32 @@ def test_error_handling_mismatch_haz_type(self):
         except Exception as e:
             self.assertEqual(str(e), "Impact calculation not possible. No impact "
                              "functions found for hazard type TC in impf_set.")
+    def test_error_handling_mismatch_impf_ids(self):
+        """Test error handling in case impf ids in exposures
+        does not appear in impf_set"""
+        haz = Hazard('TC')
+        exp = Exposures()
+        exp.gdf.loc[0,'impf_TC'] = 1
+        exp.gdf.loc[1,'impf_TC'] = 2
+        impf_exp = ImpactFunc()
+        impf_exp.id = 1
+        impf_exp.intensity = np.array([0, 20])
+        impf_exp.paa = np.array([0, 1])
+        impf_exp.mdd = np.array([0, 0.5])
+        impf_exp.haz_type = 'TC'
+        impf_noexp = deepcopy(impf_exp)
+        impf_noexp.id = 3
```

Comment:
This should work too right? Ideally, one does not first initiate an object and then modify its attributes. This is mostly a Matlab-style remnant in older CLIMADA code.

```suggestion
        impf_exp = ImpactFunc(haz_type='TC', id=1)
        impf_noexp = deepcopy(impf_exp)
        impf_noexp.id = 3
```

Reply:
Yes that's indeed better, thanks for the suggestion!